### PR TITLE
Add graftchildren! and prunebranch!

### PR DESCRIPTION
**Pull request by timholy**
This is in service of https://github.com/timholy/FlameGraphs.jl/issues/4

CC @KristofferC

### CODE REVIEW

File: src/LeftChildRightSiblingTrees.jl [hunk 1]
```diff
@@ -163,6 +165,60 @@ function depth(node::Node, d)
...
+    for c in src
+        c.parent = dest
+    end
+    if isleaf(dest)
+        dest.child = src.child
+    else
+        lastsib = lastsibling(dest.child)
+        lastsib.sibling = src.child
+    end
+    src.child = src   # make src a leaf
```
Comment (oxinabox) [inline src/LeftChildRightSiblingTrees.jl]: Is it worth making a:
```
_make_leaf!(src) = src.child = src
```
do you think?

File: src/LeftChildRightSiblingTrees.jl [hunk 2]
```diff
@@ -163,6 +165,60 @@ function depth(node::Node, d)
...
+    return dest
+end
+
+"""
+    prunebranch!(node)
+
+Eliminate `node` as a child of its parent.
```
Comment (oxinabox) [inline src/LeftChildRightSiblingTrees.jl]:
```suggestion
Eliminate `node` from the tree.
```

Comment (timholy) [reply src/LeftChildRightSiblingTrees.jl]: Good idea. I also mentioned "and all its children"

**Review by oxinabox — APPROVED** [review]

**Comment by timholy** [comment]
Thanks for the quick review, @oxinabox!

**Pull request merged**